### ISSUE TITLE
SolidJS adapter context provider may re-render children sometimes when paired with Suspense

### ISSUE THREAD

**Issue by novacrazy** [ivanhofer/typesafe-i18n#267]
### Version

5.3.4

### Describe the bug

In the SolidJS adapter, you use:
```ts
return createComponent(I18nContext.Provider, { value: ctx, children: () => props.children })
```
which defines `children` as a **function**.

This has similar behavior to:
```jsx
<Show when={...}>
{ value => ... }
</Show>
```
which will re-render the entire child tree when the given value changes. I don't fully understand exactly what changes, but as shown below it causes double-rendering.

### Reproduction

See [HERE](https://playground.solidjs.com/?hash=-166881558&version=1.3.13) for a minimal recreation (check the console to see it re-rendering components) using the same `Provider` technique as in existing typesafe-i18n.

If you instead change `children()` to `get children()` to make it a getter, SolidJS treats it much better and does not re-render.

**Comment by novacrazy** [comment]
Just got confirmation from Ryan (SolidJS creator) on Discord that using a getter is indeed the correct way, so that PR should be ideal.

**Comment by ivanhofer** [comment]
Thanks @novacrazy for investigating and providing the fix!
The issue is resolved in version `5.3.5`
